lib/ain-evm/src/executor.rs

```diff
@@ -349,6 +349,17 @@ impl<'backend> AinExecutor<'backend> {
                 contract_address,
                 direction,
             })) => {
+                // Validate nonce
+                let nonce = self.backend.get_nonce(&signed_tx.sender);
+                if nonce != signed_tx.nonce() {
+                    return Err(format_err!(
+                        "[apply_queue_tx] nonce check failed. Account nonce {}, signed_tx nonce {}",
+                        nonce,
+                        signed_tx.nonce(),
+                    )
+                    .into());
+                }
+
```

Comment:
Is this nonce check necessary? Since it is already handled in exec pipeline when we actually execute the transferdomain tx itself. Also, we dont have to worry about handling BlockSizeLimit EVM error since system tx flag is passed for DST20 bridging.